### DISCUSSION TITLE
Deleted article still appears on live version of the page (but not in draft)

### DISCUSSION THREAD

**Discussion by juliageek** [apostrophecms/apostrophe#2820] [Q&A] [answered]
I'm using:
```
"apostrophe": "^2.116.1",
"apostrophe-workflow": "^2.38.3"
```

I created an article for demoing the website I am building, and I went ahead and committed it with `apostrophe-workflow` and then committed the page it belongs to (as my project has a multiple blog structure). Then, I wanted to demo how to remove an article from the page, so I deleted the mock article. In the draft version, the article disappeared, but in the live version, it didn't.

I tried even unpublishing the article before and then batch trashing it, but the result was the same.

Then, I remembered I saw this discussion on the old ApostropheCMS forum: https://forum.apostrophecms.org/t/pages-dont-delete-or-update/788

I wonder if there is an issue opened for this? Did it get resolved? If not, what can I do to get it solved?

**Comment by boutell** [comment, marked as answer]
You need to commit the trashed article. You can use the filters to find it
in the trash, edit it and commit it.

I know this is counterintuitive, but it would not be good if there was no
"commit" step for this action.

On Mon, Mar 15, 2021 at 2:26 PM juli4g33k ***@***.***> wrote:

> I'm using:
>
> "apostrophe": "^2.116.1",
> "apostrophe-workflow": "^2.38.3"
>
> I created an article for demoing the website I am building, and I went
> ahead and committed it with apostrophe-workflow and then committed the
> page it belongs to (as my project has a multiple blog structure). Then, I
> wanted to demo how to remove an article from the page, so I deleted the
> mock article. In the draft version, the article disappeared, but in the
> live version, it didn't.
>
> I tried even unpublishing the article before and then batch trashing it,
> but the result was the same.
>
> Then, I remember I saw this discussion on the old ApostropheCMS forum:
> https://forum.apostrophecms.org/t/pages-dont-delete-or-update/788
>
> I wonder if there is an issue opened for this? Did it get resolved? If
> not, what can I do to get it solved?
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/apostrophecms/apostrophe/discussions/2820>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAAH27OL654KSRIGUTBQ4XTTDZGPFANCNFSM4ZHCWDMQ>
> .
>


-- 

THOMAS BOUTELL | CHIEF TECHNOLOGY OFFICER
APOSTROPHECMS | apostrophecms.com | he/him/his

**Reply by juliageek** [reply]
@boutell Thanks! It worked. I kind of understand the reasoning behind (I suppose you want to prevent people from deleting articles and then changing their minds, I guess), but you are right: it's counter intuitive.

**Comment by boutell** [comment]
(In A3 we are removing the UX around this.)

On Mon, Mar 15, 2021 at 4:17 PM Tom Boutell ***@***.***> wrote:

> You need to commit the trashed article. You can use the filters to find it
> in the trash, edit it and commit it.
>
> I know this is counterintuitive, but it would not be good if there was no
> "commit" step for this action.
>
> On Mon, Mar 15, 2021 at 2:26 PM juli4g33k ***@***.***>
> wrote:
>
>> I'm using:
>>
>> "apostrophe": "^2.116.1",
>> "apostrophe-workflow": "^2.38.3"
>>
>> I created an article for demoing the website I am building, and I went
>> ahead and committed it with apostrophe-workflow and then committed the
>> page it belongs to (as my project has a multiple blog structure). Then, I
>> wanted to demo how to remove an article from the page, so I deleted the
>> mock article. In the draft version, the article disappeared, but in the
>> live version, it didn't.
>>
>> I tried even unpublishing the article before and then batch trashing it,
>> but the result was the same.
>>
>> Then, I remember I saw this discussion on the old ApostropheCMS forum:
>> https://forum.apostrophecms.org/t/pages-dont-delete-or-update/788
>>
>> I wonder if there is an issue opened for this? Did it get resolved? If
>> not, what can I do to get it solved?
>>
>> —
>> You are receiving this because you are subscribed to this thread.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/apostrophecms/apostrophe/discussions/2820>, or
>> unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/AAAH27OL654KSRIGUTBQ4XTTDZGPFANCNFSM4ZHCWDMQ>
>> .
>>
>
>
> --
>
> THOMAS BOUTELL | CHIEF TECHNOLOGY OFFICER
> APOSTROPHECMS | apostrophecms.com | he/him/his
>


-- 

THOMAS BOUTELL | CHIEF TECHNOLOGY OFFICER
APOSTROPHECMS | apostrophecms.com | he/him/his

**Comment by boutell** [comment]
It is more that when we say "nothing changes on the live site until you say
'yes really,'" we mean it (:

On Mon, Mar 15, 2021 at 4:51 PM juli4g33k ***@***.***> wrote:

> @boutell <https://github.com/boutell> Thanks! It worked. I kind of
> understand the reasoning behind (I suppose you want to prevent people from
> deleting articles and then changing their minds, I guess), but you are
> right: it's counter intuitive.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/apostrophecms/apostrophe/discussions/2820#discussioncomment-485885>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAAH27O6D523H7724ABXK73TDZXM7ANCNFSM4ZHCWDMQ>
> .
>


-- 

THOMAS BOUTELL | CHIEF TECHNOLOGY OFFICER
APOSTROPHECMS | apostrophecms.com | he/him/his

**Comment by boutell** [comment]
Also consider the person who has permission to submit but not commit
changes.

But it's definitely not great UX right now.

On Mon, Mar 15, 2021 at 5:14 PM Tom Boutell ***@***.***> wrote:

> It is more that when we say "nothing changes on the live site until you
> say 'yes really,'" we mean it (:
>
> On Mon, Mar 15, 2021 at 4:51 PM juli4g33k ***@***.***>
> wrote:
>
>> @boutell <https://github.com/boutell> Thanks! It worked. I kind of
>> understand the reasoning behind (I suppose you want to prevent people from
>> deleting articles and then changing their minds, I guess), but you are
>> right: it's counter intuitive.
>>
>> —
>> You are receiving this because you were mentioned.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/apostrophecms/apostrophe/discussions/2820#discussioncomment-485885>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/AAAH27O6D523H7724ABXK73TDZXM7ANCNFSM4ZHCWDMQ>
>> .
>>
>
>
> --
>
> THOMAS BOUTELL | CHIEF TECHNOLOGY OFFICER
> APOSTROPHECMS | apostrophecms.com | he/him/his
>


-- 

THOMAS BOUTELL | CHIEF TECHNOLOGY OFFICER
APOSTROPHECMS | apostrophecms.com | he/him/his
